src/main/java/UserInterface.java

```diff
@@ -0,0 +1,223 @@
+        } while (!commandInput.equals(COMMAND_EXIT));
+    }
+
+    /**
+     * Checks the String inputted by the user and executes the appropriate command
+     * using a switch statement.
+     * @param nextLine The command to be executed.
+     */
+    private void executeCommand(String nextLine) {
+        try {
+            ArrayList<String> pieces = new ArrayList<>(Arrays.asList(nextLine.split(" ")));
+            switch (pieces.get(0)) {
+            case COMMAND_EXIT:
+                printGoodbye();
+                break;
+            case COMMAND_LIST:
+                listTasks();
+                break;
+            case COMMAND_MARK:
+                doTask(pieces.get(1));
+                break;
+            case COMMAND_UNMARK:
+                undoTask(pieces.get(1));
+                break;
+            case COMMAND_TODO:
+            case COMMAND_DEADLINE:
+            case COMMAND_EVENT:
+                addTask(pieces);
+                break;
+            default:
+                System.out.println("Command not found: " + pieces.get(0));
+            }
+        } catch (Exception e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Stores a task in the list of tasks.
+     * @param inputTasks ArrayList<String> containing input from stdin. The first String in task should represent the type of
+     *                   Task that must be added to the list of Tasks. Supported tasks: todo, deadline, event
+     */
+    private void addTask(ArrayList<String> inputTasks) {
+        try {
+            printDivider();
+            this.tasks.add(buildTask(inputTasks));
+            System.out.println("Got it. I've added this task:\n  " + this.tasks.get(this.tasks.size() - 1));
+            System.out.println("Now you have " + tasks.size() + (tasks.size() == 1 ? " task" : " tasks") + " in the list.");
+            printDivider();
+        } catch (Exception e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Constructs a subclass of Task from an ArrayList<String>.
+     * @param taskStringArray the ArrayList<String> containing the user's input, split up by whitespace.
+     *                        The first element of this list should contain the subclass of task to be constructed.
+     * @return a subclass of Task determined by the first element of taskStringArray.
+     */
+    private Task buildTask(ArrayList<String> taskStringArray) {
+        String taskType = taskStringArray.get(0);
+        ArrayList<StringBuilder> taskParts = splitTask(taskStringArray);
+        String prepositions = String.valueOf(taskParts.get(taskParts.size() - 1));
+        switch (taskType) {
+        case COMMAND_TODO:
+            return new ToDo(String.valueOf(taskParts.get(0)));
+        case COMMAND_DEADLINE:
+            return new Deadline(String.valueOf(taskParts.get(0)), String.valueOf(taskParts.get(1)), prepositions);
```

Comment:
Might be too long, can consider splitting into 2 lines

Reply:
Thank you. I will implement this.